core/src/createTransfer.ts

```diff
@@ -96,11 +89,7 @@ async function createSystemInstruction(
     connection: Connection
 ): Promise<TransactionInstruction> {
     // Check that the sender and recipient accounts exist
-    const senderInfo = await connection.getAccountInfo(sender);
-    if (!senderInfo) throw new CreateTransferError('sender not found');
-
-    const recipientInfo = await connection.getAccountInfo(recipient);
-    if (!recipientInfo) throw new CreateTransferError('recipient not found');
+    const {senderInfo, recipientInfo} = await checkAccount(connection, sender, recipient);
```

Comment:
Hmm, this refactoring adds abstraction and doesn't really save much code. In addition, here you use the return values, and in the other function, you discard them. This suggests the abstraction isn't a great fit.

Overall, what we probably _really_ want to refactor to is something that expresses this:

- For system transfers, check to make sure the sender exists†, and _maybe_ check to make sure the recipient exists (the argument against is that zero balance should probably be allowed).
- For token transfers, check to make sure the _sender's ATA_ exists, and also check to to make sure the _recipient's ATA_ exists. We don't actually care if the sender† or recipient _System_ accounts exist or have SOL.

† We might also want to have a check for the SOL balance of the sender, since the sender is the expected fee payer. Probably this would use `Connection#getFeeForMessage` rather than hardcoding 5000 lamports as the cost. For System transfers, this will need to be calculated in addition to the transfer amount.